src/TestingFramework/TestCase/FunctionalTestCase.php

```diff
@@ -111,6 +111,13 @@
      */
     protected $configurationToUseInTestInstance = [];
 
+    /**
+     * If true, deprecation errors are not handled.
+     *
+     * @var bool
+     */
+    protected $disableDeprecations = false;
```

Comment:
I propose removing the negative, i.e., changing it to "enableDeprecations" (with a default value of true). Enabling a flag to disable something is confusing and error prone.

Reply:
I'll do my best. :-)

You can see the "cofusingness" as a combination of "number of mental steps needed to understand" and "level of contradiction"/"intuitiveness"

Number of mental steps:
- not reversed: 1. enable (flag) to 2. enable (feature)
- reversed: 1. enable (flag) to 2. (not) 3. enable (feature)

Level of contradiction:
- not reversed: flag enabled = feature enabled, flag disabled = feature disabled
- reversed: flag enabled = feature disabled, flag disabled = feature enabled

This also has to do with the mind grasping a statement first (intutively), and having trouble directly integrating a negation. For example, people get stressed out far more if someone tells them "I don't want to tell you that there is a problem." vs. "I want to tell you that everything works fine."

For background, I recommend this article:
https://schneide.wordpress.com/2014/08/03/dont-ever-not-avoid-negative-logic/

Reply:
And another article on the psychological aspects of negations in general (not programming-related): https://www.speechcare.de/news/wie-verneinungen-unser-gehirn-verwirren/

Reply:
Hi @oliverklee,

Thanks for sharing your point and the article!